lib/tmp.js

```diff
@@ -196,9 +196,20 @@ function _createTmpFile(options, callback) {
       if (err) return cb(err);
 
       var removeCallback = _prepareRemoveCallback(function(fdPath) {
-        fs.closeSync(fdPath[0]);
-        fs.unlinkSync(fdPath[1]);
-      }, [fd, name]);
+        try {
+          fs.closeSync(fdPath[0]);
+        }
+        catch (e) {
+          // did user close the fd?
+          if (Math.abs(e.errno) != _c.EBADF) {
```

Comment:
The Math.abs is needed because node will make the constant.EBADF a negative number upon throwing the exception.

Of course we could simply add the two numbers and see whether it turns out to be 0...